cloud/disk_manager/internal/pkg/common/inflight_queue.go

```diff
@@ -150,21 +164,27 @@ func (q *InflightQueue) valueProcessed(value uint32) {
 		}
 	}
 
-	if toRemoveCount == 0 {
-		return
-	}
+	q.updateMilestoneOnDrain(toRemoveCount)
+
+	// Remove processed (not in-flight) items from the head.
+	q.items = q.items[toRemoveCount:]
+}
+
+func (q *InflightQueue) updateMilestoneOnDrain(toRemoveCount int) {
+	newMilestoneValue := q.milestone.Value
 
-	newMilestoneValue := uint32(0)
 	if toRemoveCount >= len(q.items) {
-		lastItemValue := q.items[len(q.items)-1].value
-		newMilestoneValue = lastItemValue + 1
+		if len(q.items) > 0 {
```

Comment:
Can we handle `len(q.items) == 0` case separately? Above those if https://github.com/ydb-platform/nbs/pull/1065/files#diff-48919f6822c8e44e648356b9db6f6ca0735fa72904becf63e8372fa4930894aeL158

Reply:
Hmm, maybe we should not check the case len(q.items) == 0? Because this can not be true: if we have a processed value, this value should be present in queue.